src/NHibernate/Proxy/StaticProxyFactory.cs

```diff
@@ -54,7 +54,21 @@ public override INHibernateProxy GetProxy(object id, ISessionImplementor session
 				GetIdentifierMethod,
 				SetIdentifierMethod,
 				ComponentIdType);
-			_cacheEntry = new ProxyCacheEntry(IsClassProxy ? PersistentClass : typeof(object), Interfaces);
+			_cacheEntry = new ProxyCacheEntry(GetProxyBaseType(PersistentClass), Interfaces);
+		}
+
+		private static System.Type GetProxyBaseType(System.Type type)
+		{
+			if (type.IsClass)
+			{
+				for (var t = type; t != null; t = t.BaseType)
```

Comment:
It seems `session.Load<Cat>(catId)` test is going to fail If modify `Cat` by adding some unmapped base class.

Reply:
No she'll be fine.

Reply:
No, she'll be right.